artemis/modules/moodle_scanner.py

```diff
@@ -0,0 +1,149 @@
+        super().__init__(*args, **kwargs)
+
+    def run(self, current_task: Task) -> None:
+        base_url = get_target_url(current_task)
+        self.log.info(f"Starting moodlescan for {base_url}")
+
+        try:
+            # Run moodlescan with error output captured
+            process = subprocess.run(
+                ["python3", "moodlescan.py", "-u", base_url, "-r", "-k"],
+                cwd="/moodle_scanner",
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+
+            self.log.info(f"Moodlescan stdout: {process.stdout}")
+            if process.stderr:
+                self.log.warning(f"Moodlescan stderr: {process.stderr}")
+
+            # Parse the output for relevant information
+            output_lines = process.stdout.splitlines()
+            server_info = None
+            version_info = None
+            vulnerabilities = []
+
+            for i, line in enumerate(output_lines):
+                if "Error: Can't connect" in line:
+                    self.log.info(f"Connection error: {line}")
+                    self.db.save_task_result(
+                        task=current_task, status=TaskStatus.OK, status_reason=line, data={"raw_output": process.stdout}
+                    )
+                    return
+
+                if "server" in line.lower() and ":" in line:
+                    server_info = line.split(":", 1)[1].strip()
+                elif "version" in line.lower() and not line.startswith("."):
+                    # Look at next line for version info if it's not dots or error
+                    if i + 1 < len(output_lines):
+                        next_line = output_lines[i + 1].strip()
+                        if next_line and not next_line.startswith(".") and "error" not in next_line.lower():
+                            version_info = next_line
+                elif "vulnerability" in line.lower() or "cve" in line.lower():
+                    vulnerabilities.append(line.strip())
+
+            result = {
+                "server": server_info,
+                "version": version_info,
+                "vulnerabilities": vulnerabilities,
+                "raw_output": process.stdout,
+            }
+
+            # Determine if anything interesting was found
+            if vulnerabilities:
+                status = TaskStatus.INTERESTING
+                status_reason = f"Found: {', '.join(vulnerabilities)}"
+            elif version_info and version_info != "Version not found":
+                status = TaskStatus.INTERESTING
+                status_reason = f"Found version: {version_info}"
+            else:
+                status = TaskStatus.OK
+                status_reason = "Version not found" if version_info == "Version not found" else None
+
+            self.db.save_task_result(task=current_task, status=status, status_reason=status_reason, data=result)
+
+        except subprocess.CalledProcessError as e:
```

Comment:
the try...except block may wrap only the subprocess call in lines 77...84, right?